lib/bolt/inventory/inventory2.rb

```diff
@@ -136,6 +136,18 @@ def expand_targets(targets)
       end
       private :expand_targets
 
+      def remove_target(current_group, target, desired_group)
+        if current_group.name == desired_group
+          current_group.remove_target(target)
+        end
+        current_group.groups.each do |child_group|
+          # If target was in current group, remove it from all child groups
+          desired_group = child_group.name if current_group.name == desired_group
```

Comment:
This looks like it will only work for the first child, since after that we've overridden `desired_group`. Should we have a separate variable for this?

Reply:
Ah, good catch. Updated.